enclave/sgx/verifier.c

```diff
@@ -0,0 +1,308 @@
+    const void* p_tcb_info_issuer_chain,
+    uint32_t tcb_info_issuer_chain_size,
+    const void* p_pck_crl,
+    uint32_t pck_crl_size,
+    const void* p_root_ca_crl,
+    uint32_t root_ca_crl_size,
+    const void* p_pck_crl_issuer_chain,
+    uint32_t pck_crl_issuer_chain_size,
+    const void* p_qe_identity,
+    uint32_t qe_identity_size,
+    const void* p_qe_identity_issuer_chain,
+    uint32_t qe_identity_issuer_chain_size)
+{
+    OE_UNUSED(format_id);
+    OE_UNUSED(opt_params);
+    OE_UNUSED(opt_params_size);
+    OE_UNUSED(p_quote);
+    OE_UNUSED(quote_size);
+    OE_UNUSED(expiration_check_date);
+    OE_UNUSED(p_collateral_expiration_status);
+    OE_UNUSED(p_quote_verification_result);
+    OE_UNUSED(p_qve_report_info);
+    OE_UNUSED(qve_report_size);
+    OE_UNUSED(p_supplemental_data);
+    OE_UNUSED(supplemental_data_size);
+    OE_UNUSED(p_supplemental_data_size_out);
+    OE_UNUSED(collateral_version);
+    OE_UNUSED(p_tcb_info);
+    OE_UNUSED(tcb_info_size);
+    OE_UNUSED(p_tcb_info_issuer_chain);
+    OE_UNUSED(tcb_info_issuer_chain_size);
+    OE_UNUSED(p_pck_crl);
+    OE_UNUSED(pck_crl_size);
+    OE_UNUSED(p_root_ca_crl);
+    OE_UNUSED(root_ca_crl_size);
+    OE_UNUSED(p_pck_crl_issuer_chain);
+    OE_UNUSED(pck_crl_issuer_chain_size);
+    OE_UNUSED(p_qe_identity);
+    OE_UNUSED(qe_identity_size);
+    OE_UNUSED(p_qe_identity_issuer_chain);
+    OE_UNUSED(qe_identity_issuer_chain_size);
+
+
+    if (_retval)
+        *_retval = OE_UNSUPPORTED;
+
+    return OE_UNSUPPORTED;
+}
+OE_WEAK_ALIAS(
+    _oe_verify_quote_ocall, oe_verify_quote_ocall);
+#endif
+
+
+oe_result_t oe_verify_qve_report_and_identity(
+        const uint8_t* p_quote,
+        uint32_t quote_size,
+        const sgx_ql_qe_report_info_t* p_qve_report_info,
+        time_t expiration_check_date,
+        uint32_t collateral_expiration_status,
+        uint32_t quote_verification_result,
+        const uint8_t *p_supplemental_data,
+        uint32_t supplemental_data_size,
+        uint16_t qve_isvsvn_threshold)
+{
+    oe_result_t result = OE_UNEXPECTED;
+
+    oe_sha256_context_t sha_handle = { 0 };
+    OE_SHA256 report_data_hash = { 0 };
+    sgx_report_data_t report_data = { 0 };
+
+    if (p_quote == NULL ||
+            p_qve_report_info == NULL ||
+            sizeof(*p_qve_report_info) != sizeof(sgx_ql_qe_report_info_t) ||
+            !oe_is_within_enclave(p_quote, quote_size) ||
+            !oe_is_within_enclave(p_qve_report_info, sizeof(sgx_ql_qe_report_info_t)) ||
+            (p_supplemental_data == NULL && supplemental_data_size != 0) ||
+            (p_supplemental_data != NULL && supplemental_data_size == 0))
+        return OE_INVALID_PARAMETER;
+
+    if (p_supplemental_data && supplemental_data_size > 0) {
+        if (!oe_is_within_enclave(p_supplemental_data, supplemental_data_size)) {
+            return OE_INVALID_PARAMETER;
+        }
+    }
+
+    //Defense in depth, threshold must be greater or equal to 3
+    //
```

Comment:
```suggestion
    // Defense in depth, threshold must be greater or equal to 3
```
nit: Also fix comment style in 217 etc.

Reply:
Fixed.